src/coreclr/vm/gcheaputilities.cpp

```diff
@@ -41,7 +41,11 @@ bool g_sw_ww_enabled_for_gc_heap = false;
 
 #endif // FEATURE_USE_SOFTWARE_WRITE_WATCH_FOR_GC_HEAP
 
-GVAL_IMPL_INIT(gc_alloc_context, g_global_alloc_context, {});
+CLRRandom m_random; // used to randomize allocation sampling
+CLRRandom* GetRandomizer() { return &m_random;}
```

Comment:
This does not look thread safe.

CLRRandom is not thread safe. Using it from multiple threads may corrupt its state and it can start returning degraded random numbers or even a constant.

Reply:
Good point...
However, I'm not sure that this PR is worth implementing [Xorshiro](https://github.com/dotnet/runtime/blob/ce1477b5df9b378f1dfcbb9194aeed3932ecfc03/src/libraries/System.Private.CoreLib/src/System/Random.Xoshiro128StarStarImpl.cs#L31) in C and have it per thread. Would it be possible to keep the original ClrRandom per thread?

Reply:
ClrRandom per thread is fine. I should be allocated lazily (https://github.com/dotnet/runtime/pull/100356#discussion_r1614246680).